---
title: "Test run for SNP-based GS"
output: NULL
author: Jinliang Yang
---


/common/jyanglab/gxu6/project/07_282panel_GCTB/genotype


```{r}
library("huskeR")

trainfile <- list.files(path="/common/jyanglab/jyang21/projects/Kmer-GS/largedata/nam/", 
                        pattern="train", full.names=TRUE)
testfile <- list.files(path="/common/jyanglab/jyang21/projects/Kmer-GS/largedata/nam/", 
                      pattern="test", full.names=TRUE)

inputdf <- data.frame(pi=0.995,
   geno="/common/jyanglab/jyang21/projects/Kmer-GS/largedata/genotype/raw_snp0.7M.txt.newbin",
   trainpheno=trainfile,
   testpheno=testfile,
   chainLength=41000, burnin=1000, varGenotypic=2, varResidual=1,
   out=gsub("_train.txt|.*\\/", "", trainfile))

run_GenSel4(inputdf, cv=TRUE, 
            inpdir="largedata/nam/", 
            cmdno=1,
            shid = "slurm-script/run_gensel_array.sh", remove=FALSE,
            email="yangjl0930@gmail.com", 
            runinfo = c(TRUE, "jyanglab", 2, "2G", "8:00:00") )

```


## Obtain the results

```{r}
f <- list.files(path="/common/jyanglab/jyang21/projects/Kmer-GS/largedata/pheno/", 
                pattern="ghatREL", full.names=TRUE)

library("data.table")

out <- data.frame()
for(i in 1:length(f)){
   df <- fread(f[i], data.table=FALSE)
   r <- cor(df$gHat, df$DTS, method="pearson")
   tem <- data.frame(r=r, file=f[i])
   out <- rbind(out, tem)
}

```


